pandas/_libs/internals.pyx

```diff
@@ -105,7 +107,9 @@ cdef class BlockPlacement:
             Py_ssize_t start, stop, end, _
         if not self._has_array:
             start, stop, step, _ = slice_get_indices_ex(self._as_slice)
-            self._as_array = np.arange(start, stop, step, dtype=np.int64)
+            # NOTE: this is the C-optimized equivalent of
+            # np.arange(start, stop, step, dtype=np.int64)
```

Comment:
nitpick: i like to add an extra space at the beginning of multi-line comments specifically to avoid confusing this for commented-out code.  quotation marks or backticks or ... are also good